src/PaddedViews.jl

```diff
@@ -168,6 +168,13 @@ Base.parent(A::PaddedView) = A.data
     return A.fillvalue
 end
 
+function Base.setindex!(A::PaddedView{T, N}, v, i::Vararg{Int, N}) where {T, N}
+    @boundscheck checkbounds(A, i...)
+    @boundscheck checkbounds(A.data, i...)
```

Comment:
I can understand providing a special error message in case the first passes and the second does not, but this doesn't seem to do that clearly. Why not just the second one? Maybe something like
```julia
@boundscheck checkbounds(Bool, A.data, i...) || throw(checkbounds(Bool, A, i...) ? ArgumentError("PaddedViews do not support (re)setting the padding value. Consider making a copy of the array first.") : BoundsError(A, i))
```

Reply:
Hmm, yeah, this is clearer.

Reply:
The logic is a bit incorrect and I've updated it in 4e77ec5

The black box is the padding indices, while the orange box is the parent data indices

![image](https://user-images.githubusercontent.com/8684355/95673448-29aed880-0bdb-11eb-9b76-a9f0d0dbc116.png)